src/auction.js

```diff
@@ -397,10 +398,16 @@ export function auctionCallbacks(auctionDone, auctionInstance) {
 
   function adapterDone() {
     let bidderRequest = this;
+    let bidderRequests = auctionInstance.getBidRequests();
 
     bidderRequestsDone.add(bidderRequest);
-    allAdapterCalledDone = auctionInstance.getBidRequests()
-      .every(bidderRequest => bidderRequestsDone.has(bidderRequest));
+    allAdapterCalledDone = bidderRequests.filter(request => {
+      const auctionOptionsConfig = config.getConfig('auctionOptions');
+      if (auctionOptionsConfig && auctionOptionsConfig.secondaryBidders && !bidderRequests.every(bidder => includes(auctionOptionsConfig.secondaryBidders, bidder.bidderCode))) {
```

Comment:
Run the `bidderRequests.every` check outside of the filter function. The result should be the same each time, so now need to have this loop within a loop.

Also, just want to make sure I'm understanding this. When determining whether or not the auction is done, don't wait for secondary bidders, unless all of the bidders are secondary bidders. Is that correct?

Reply:
Thanks for the feedback. That is the correct logic. I pushed an update to fix the nested loop and address the legibility of the logic.